src/store/Workspaces/index.ts

```diff
@@ -197,20 +209,15 @@ export const actionCreators: ActionCreators = {
   },
 
   stopWorkspace: (workspaceId: string): AppThunk<KnownAction, Promise<void>> => async (dispatch): Promise<void> => {
-    dispatch({ type: 'REQUEST_WORKSPACES' });
-
     try {
-      const workspace = await WorkspaceClient.restApiClient.stop(workspaceId);
-      dispatch({ type: 'UPDATE_WORKSPACE', workspace });
```

Comment:
```dispatch({ type: 'UPDATE_WORKSPACE', workspace });``` was removed.
In this case, if something happens with WebSocket connection we will lose status changes ```
start -> stopping``` and leave a possibility to send several requests to the server-side. It will generate lots of errors.
WDYT?

Reply:
`restApiClient.stop()` returns `Promise<void>` hence we don't lose anything here.

Reply:
I disagree with you:

https://github.com/eclipse/che-workspace-client/blob/master/src/rest/remote-api.ts#L301
```
   /**
     * Stops a workspace.
     *
     * @param {string} workspaceId a workspace ID.
     * @returns {Promise<any>}
     */
    public stop(workspaceId: string): Promise<any> {
        return new Promise<any>((resolve, reject) => {
            this.remoteAPI.stop(workspaceId)
                .then((response: AxiosResponse<any>) => {
                    resolve(response.data);
                })
                .catch((error: AxiosError) => {
                    reject(new RequestError(error));
                });
        });
    }
```

Reply:
Correct me if I'm wrong, but this request successfully resolves with `204 No Content`.

https://github.com/eclipse/che/blob/b2e83ac1a411a15e45ea90f0aebf6072c0723b35/wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceService.java#L398-L414
```java
  @DELETE
  @Path("/{id}/runtime")
  @ApiOperation(
      value = "Stop the workspace",
      notes =
          "This operation can be performed only by the workspace owner."
              + "The workspace stops asynchronously")
  @ApiResponses({
    @ApiResponse(code = 204, message = "The workspace is stopping"),
    @ApiResponse(code = 404, message = "The workspace with specified id doesn't exist"),
    @ApiResponse(code = 403, message = "The user is not workspace owner"),
    @ApiResponse(code = 500, message = "Internal server error occurred")
  })
  public void stop(@ApiParam("The workspace id") @PathParam("id") String id)
      throws ForbiddenException, NotFoundException, ServerException, ConflictException {
    workspaceManager.stopWorkspace(id, emptyMap());
  }
```